README.md

```diff
@@ -1,2 +1,184 @@
+import factory from 'trampss-redux-data-store'
+export default factory('id')('api')('todos')
+```
+That's it, you exported a function wich is a reducer to include in a Redux `combineReducer`.
+The exported function also have actions and selectors ! Easy.
+
+In this example, we have a `todos` reducer, it has to be combined into `state.api.todos`
+
+## Why
+Using `redux-saga` make our Redux code base simpler : it's like a key/value store. But one drawback is the amount of duplicate code, each ressource has its reducers, actions and selectors.
+
+To avoid growing Redux code base, and to avoid inconsistency, and pain maintenability, we created this lightweight library (<4Kb) that is a factory of reducers, actions and selectors.
+
+## Installation
+ - `yarn add trampss-redux-data-store`
+ - `npm i trampss-redux-data-store``
+
+### peer dependency
+ - `lodash` : we use the minimum of lodash function trying to have a lightweight webpack bundle.
+   - `keyBy`
+   - `without`
+   - `uniq`
+   - `omit`
+   - `at`
+
+## API
+ - [factory](#factory)
+ - [actions](#actions)
+ - [selectors](#selectors)
+
+### factory
+You need to use the factory to get a new set of reducer/actions/selectors :
+```es6
+import factory from 'trampss-redux-data-store'
+```
+
+This factory takes three parameters :
+ - **first**, the field used to identify your objects (`id` for example)
+   - you have to set this parameter.
+ - **second**, where the reducer will be combined via `combineReducer`
+   - you can leave it empty: the reducer is to the root of the Redux state
+   - you can use dot notation, like `api.raw`: your reducer will be combined into `state.api.raw.<your_reducer>`
+ - **third**, the reducer name (for instance: `todos`)
+   - it's used to generate actions types
+   - it's used to retrieve informations from selectors
+
+Example:
+ - this reducer will use `id` as key field
+ - it's combined into `state.api.raw`
+ - its name is `todos`
+```es6
+import factory from 'trampss-redux-data-store'
+// factory(fieldKey)(path)(name)
+export default factory('id')('api.raw')('todos')
+```
+
+Data will be stored into `state.api.raw.todos`
+
+### reducer
+Previous factory returns a function which is a reducer.
+You just have to combine it like any other reducer :
+```es6
+import { createStore, combineReducers, compose, applyMiddleware } from 'redux'
+
+// import your reducer
+// (created by tramps-redux-data-store factory)
+import todos from './myTodosReducer'
+
+// create your Redux store as usual
+const store = createStore(
+  combineReducers({
+    api: combineReducers({
+      raw: combineReducers({
+        // import your reducer into api.raw
+        // since we configured this path
+        todos,
+      }),
+    }),
+  }),
+  /* your Redux middlewares */
+)
+
+export default store
+```
+
+### actions
+The factory returns a reducer (function). It also contains actions and selectors as fields.
+We have created some generic actions. By now, this is not possible to add yours.
```

Comment:
Some generic actions are available. By now, it's not possible to add custom ones